### ISSUE TITLE
`turbo gen run` throws `error TS6053: File '@tsconfig/node20/tsconfig.json' not found.`

### ISSUE THREAD

**Issue by lbdm44** [vercel/turbo#7502]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

https://github.com/lbdm44/tsconfig-node20-turbo-gen-issue

### What package manager are you using / does the bug impact?

Yarn v2/v3/v4 (node_modules linker only)

### What operating system are you using?

Mac

### Which canary version will you have in your reproduction?

1.12.5-canary.0

### Describe the Bug

When trying to run `yarn turbo gen run`, an error is thrown:

```sh
>>> Modify "tsconfig-node20-turbo-gen-issue" using custom generators


>>> Unexpected error. Please report it as a bug:
error TS6053: File '@tsconfig/node20/tsconfig.json' not found.
```

This happens only when using [@tsconfig/node20](https://www.npmjs.com/package/@tsconfig/node20) and specifying a version of `typescript` in `package.json`.

### Expected Behavior

Expected behavior is that the generator runs without erroring out.

### To Reproduce

1. Check out https://github.com/lbdm44/tsconfig-node20-turbo-gen-issue
2. `yarn install`
3. `yarn turbo gen run`

**Comment by tknickman** [comment]
hey @lbdm44, can you see if this solves your issue? 

https://github.com/vercel/turbo/issues/6656#issuecomment-1834300667

**Comment by tknickman** [comment]
Closing because it _looks_ like this is the same issue, but if something else is going on here feel free to re-open!

**Comment by zifahm** [comment]
I started getting the error with typescript: 5.4.2. Fixed it by pining typescript to 5.3.3
